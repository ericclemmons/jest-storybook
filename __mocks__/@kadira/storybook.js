import renderer from "react-test-renderer";

export const action = (actionName) => jest.fn();

export const storiesOf = (groupName) => {
  const api = {
    add(storyName, story) {
      describe(groupName, () => {
        it(storyName, () => {
          const component = renderer.create(story());

          // @TODO Enable custom names once released:
          // > https://github.com/facebook/jest/pull/2094
          expect(component.toJSON()).toMatchSnapshot(
            // `${groupName}.${storyName}`
          );
        });
      });

      return api;
    },

    addDecorator() {
      return api;
    },
  };

  return api;
};
