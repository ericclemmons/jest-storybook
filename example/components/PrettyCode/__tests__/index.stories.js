import React from "react";
import { storiesOf, action } from "@kadira/storybook";

import PrettyCode  from "../";

storiesOf("PrettyCode", module)
  .add("defaults to JavaScript", () => (
    <PrettyCode>{`
      const language = "JavaScript";

      export default "Howdy!";
    `}</PrettyCode>
  ))
  .add("supports CSS", () => (
    <PrettyCode language="css">{`
      article {
        color: #000;
        background: white;
      }

      h1 {
        font-weight: bold;
      }
    `}</PrettyCode>
  ))
  .add("supports HTML", () => (
    <PrettyCode language="html">{`
      <article>
        <h1>HTML</h1>
        <p>Howdy!</p>
      </article>
    `}</PrettyCode>
  ))
  .add("supports JSX", () => (
    <PrettyCode language="jsx">{`
      ReactDOM.render((
        <article>
          <h1>HTML</h1>
          <p>Howdy!</p>
        </article>
      ), window.app);
    `}</PrettyCode>
  ))
  .add("supports Markdown (GFM)", () => (
    <PrettyCode language="markdown">{`
      Markdown
      ========

      > _Howdy!_

      ${"```"}js
      const language = "JavaScript";

      export default "Howdy!";
      ${"```"}
    `}</PrettyCode>
  ))
;
