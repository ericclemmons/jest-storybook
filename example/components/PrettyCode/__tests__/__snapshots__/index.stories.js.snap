exports[`PrettyCode defaults to JavaScript 1`] = `
<div>
  <code>
    mock-react-codemirror
  </code>
  <pre>
    {
      \"options\": {
        \"mode\": \"jsx\",
        \"readOnly\": true,
        \"theme\": \"monokai\"
      },
      \"value\": \"const language = \\\"JavaScript\\\";\\n\\nexport default \\\"Howdy!\\\";\"
    }
  </pre>
</div>
`;

exports[`PrettyCode supports CSS 1`] = `
<div>
  <code>
    mock-react-codemirror
  </code>
  <pre>
    {
      \"options\": {
        \"mode\": \"css\",
        \"readOnly\": true,
        \"theme\": \"monokai\"
      },
      \"value\": \"article {\\n  color: #000;\\n  background: white;\\n}\\n\\nh1 {\\n  font-weight: bold;\\n}\"
    }
  </pre>
</div>
`;

exports[`PrettyCode supports HTML 1`] = `
<div>
  <code>
    mock-react-codemirror
  </code>
  <pre>
    {
      \"options\": {
        \"mode\": \"htmlmixed\",
        \"readOnly\": true,
        \"theme\": \"monokai\"
      },
      \"value\": \"<article>\\n  <h1>HTML</h1>\\n  <p>Howdy!</p>\\n</article>\"
    }
  </pre>
</div>
`;

exports[`PrettyCode supports JSX 1`] = `
<div>
  <code>
    mock-react-codemirror
  </code>
  <pre>
    {
      \"options\": {
        \"mode\": \"jsx\",
        \"readOnly\": true,
        \"theme\": \"monokai\"
      },
      \"value\": \"ReactDOM.render((\\n  <article>\\n    <h1>HTML</h1>\\n    <p>Howdy!</p>\\n  </article>\\n), window.app);\"
    }
  </pre>
</div>
`;

exports[`PrettyCode supports Markdown (GFM) 1`] = `
<div>
  <code>
    mock-react-codemirror
  </code>
  <pre>
    {
      \"options\": {
        \"mode\": \"gfm\",
        \"readOnly\": true,
        \"theme\": \"monokai\"
      },
      \"value\": \"Markdown\\n========\\n\\n> _Howdy!_\\n\\n\`\`\`js\\nconst language = \\\"JavaScript\\\";\\n\\nexport default \\\"Howdy!\\\";\\n\`\`\`\"
    }
  </pre>
</div>
`;
